cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -157,12 +159,15 @@ public class AppMetadataStore {
   /**
    * Static method for creating an instance of {@link AppMetadataStore}.
    */
-  public static AppMetadataStore create(StructuredTableContext context) {
-    return new AppMetadataStore(context);
+  public static AppMetadataStore create(StructuredTableContext context,
```

Comment:
It is better to keep the existing constructor and add an overloaded one that takes two arguments when it is used at multiple places. The existing one then can pass null to the overloaded one. This will remove the need to change all callers to pass a null/default argument.

Reply:
I agree, however I believe earlier had a conversation with @albertshau @chtyim that it is not a good practice to overload constructors. @albertshau Would it be fine to have one here, since it simplifies the code?

Reply:
The context of the earlier conversation was in a class where there were something like 8 different constructors. At that point, using the Builder pattern makes more sense. If it's just a couple constructors it's not a big deal.